packages/lib/computer.d.ts

```diff
@@ -1,377 +1,558 @@
-}
-interface Utxo {
-  hash: any;
-  index: any;
-  nonWitnessUtxo: Buffer;
-}
-interface Effect {
-  res: Json;
-  env: JObject;
+  txId: string
+  txHex: string
+  vsize: number
+  version: number
+  locktime: number
+  ins: _Input[]
+  outs: _Output[]
+}
+type StringOrArray<T> = string | Array<StringOrArray<T>>
+type Utxo = {
+  hash: any
+  index: any
+  nonWitnessUtxo: Buffer
+}
+type Effect = {
+  res: Json
+  env: JObject
+}
+
+class RestClient {
+  readonly chain: TBCChain
+  readonly network: TBCNetwork
+  readonly networkObj: Network
+  readonly mnemonic: string
+  readonly path: string
+  readonly passphrase: string
+  readonly addressType: AddressType
+  readonly keyPair: BIP32Interface
+  readonly bcn: UrlFetch
+  readonly dustRelayTxFee: number
+  readonly moduleStorageType: ModuleStorageType
+  satPerByte: number
+  constructor({
+    chain,
+    network,
+    mnemonic,
+    path,
+    passphrase,
+    addressType,
+    url,
+    satPerByte,
+    dustRelayFee,
+    moduleStorageType,
+  }?: ComputerOptions)
+  rpc(method: string, params: string): Promise<any>
+  broadcast(txHex: string): Promise<string>
+  getBalance(address: string): Promise<_Balance>
+  listTxs(address: string): Promise<_Transaction>
+  getUtxos(address: string): Promise<_Unspent[]>
+  getFormattedUtxos(address: string): Promise<_Unspent[]>
+  getRawTxs(txIds: string[]): Promise<string[]>
+  getTx(txId: string): Promise<_Transaction>
+  query({ publicKey, limit, offset, order, ids, mod }: Partial<Query>): Promise<string[]>
+  idsToRevs(outIds: string[]): Promise<string[]>
+  revToId(rev: string): Promise<string>
+  static getSecretOutput({ _url, keyPair }: { _url: string; keyPair: BIP32Interface }): Promise<{
+    host: string
+    data: string
+  }>
+  static setSecretOutput({
+    secretOutput,
+    host,
+    keyPair,
+  }: {
+    secretOutput: SecretOutput
+    host: string
+    keyPair: BIP32Interface
+  }): Promise<Data & (Encrypted | Stored)>
+  static deleteSecretOutput({
+    _url,
+    keyPair,
+  }: {
+    _url: string
+    keyPair: BIP32Interface
+  }): Promise<void>
+  faucet(address: string, value: number): Promise<_Unspent>
+  faucetScript(output: Buffer, value: number): Promise<_Unspent>
+  mine(count: number): Promise<void>
+  verify(txo: _Unspent): Promise<void>
+  height(): Promise<number>
+}
+
+class Wallet {
+  readonly restClient: RestClient
+  constructor(params?: ComputerOptions)
+  derive(subpath?: string): Wallet
+  getBalance(address?: string): Promise<_Balance>
+  getUtxos(address?: string): Promise<_Unspent[]>
+  getDustThreshold(isWitnessProgram: boolean, script?: Buffer): number
+  getAmountThreshold(script: Buffer, isWitnessProgram?: boolean): number
+  getUtxosWithOpts({ include, exclude }?: FundOptions): Promise<_Unspent[]>
+  fetchUtxo: ({ txId, vout }: _Unspent) => Promise<Utxo>
+  checkFee(fee: number, size: number): void
+  getSigOpCount(script: Buffer): number
+  getLegacySigOpCount(tx: Transaction): Promise<number>
```

Comment:
For the wallet, we also should use TBCTransaction instead of nakamotoJs transactions.